### PR TITLE
[WIP] Deeper actor integration; blockchain policy prep

### PR DESCRIPTION
**Pull request by KPrasch**
Aims to prepare nucypher for a blockchain-backed `alice.grant`.

Most notably, characters can be instantiated (given that a `NucypherConfig` class is injected), a blockchain connection will be established through the new `Interface` API, and, optionally, the appropriate agents are created through the `ContractDeployer`'s agency, providing characters direct access to contract methods. 

The Initial connection to a blockchain is the population of `Blockchain._instance` singelton, and occurs in `NucypherConfig.__init__`; called when creating a character. 

Depends on PR #276 

tldr; Blockchain-Characters can be created with a working config, unfaked `dht_key` publication, ether_address, contract methods etc. 

- Reduces complexity of `ContractAgent` API with default values
- Compose `NucypherConfig` and `BlockchainConfig`
- Improves blockchain singelton usage in config
- Refactors `NucypherMinerConstants` into a module that uses constant sorrow + side effects
- Implement co-op inheritance scheme 
- Deprecates `NuCypherDHTNode` and `Capabilities`
- Renames `NetworkyStuff` -> `NetworkMiddleware`
- Cleans up unused code

### CODE REVIEW

**Comment by KPrasch** [comment]
Moved to PR #286!
